### DISCUSSION TITLE
VIM Tiny or Nano editor.

### DISCUSSION THREAD

**Discussion by pcho** [blinksh/blink#1515] [Ideas]
Not everyone needs VS Code, and there's not a small group of users who would benefit significantly from having VIM or Nano on board. Also, VS Code doesn't work without an internet connection. :)

**Comment by carloscabanero** [comment]
We usually decide on ideas based on what we think the “core of Blink” should be. One thing we assume on our users is that they are always connected. We want to facilitate connected flows because that is where we think the future should be. Within this framework, there are features that fit, like #1514, and others that right now may not - although that may change in the future. 

Truth is the base UNIX set of tools is barebones on purpose, because it is also hard to maintain. Ios_system is a huge achievement, but we and even Nicholas are investigating better ways to do things there.

**Reply by kevinrook** [reply]
Any update on maybe adding Vi Vim Nano to Blink for local shell editing ?

**Reply by carloscabanero** [reply]
The way UNIX tools work right now with Blink are not up to my standard. They all have their quirks and I apologize for that. 

My plan is to dedicate some time next month to see what can be done and what is sustainable for the project. We are focused on making the remote feel like local, so that's our priority. If we can stabilize the local tools and maybe add a few more, in a reliable and sustainable way, considering our scarce resources, then so be it. But if not, we will consider alternatives.

**Reply by carloscabanero** [reply]
Done on 17.3.0
